lib/shopify_cli/theme/dev_server/dev_headers.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module ShopifyCLI
+  module Theme
+    module DevServer
+      class DevHeaders
+        BLOCKED_HEADERS = %w(content-security-policy)
```

Comment:
Could you simply add this to https://github.com/Shopify/shopify-cli/blob/main/lib/shopify_cli/theme/dev_server/proxy.rb#L175? Seems the logic is duplicated there too, and removing `content-security-policy` is strictly related to using a proxy, so seems fitting to me.